Model/Data/AmwalOrderDetails.php

```diff
@@ -61,16 +61,12 @@ public function updateOrderStatus()
         $orderId = $requestBody['order_id'];
         $refId = $requestBody['ref_id'];
 
-        $amwalOrderData = $this->getAmwalOrderData->execute($amwalOrderId);
+        $order = $this->getOrderByAmwalOrderId($amwalOrderId, $orderId, $refId);
+        $amwalOrderData = $this->orderUpdate->update($order, 'AmwalOrderDetails', true);
+
         if (!$amwalOrderData) {
             return false;
         }
```

Comment:
To be more consistent with what we return we would consider returning true on success